xilem/src/lib.rs

```diff
@@ -38,6 +38,16 @@ where
     driver: MasonryDriver<State, Logic, View, View::ViewState>,
 }
 
+// This struct is same as the Xilem struct, except it exposes members as public but implements
+// no methods. It is used to split Xilem into parts for use with external event loops.
+pub struct XilemParts<State, Logic, View>
```

Comment:
What's the reasoning for having this struct? The other options I can see are:
- Returning a tuple
- Making the fields on `Xilem` public

Reply:
Yeah, initially I returned a tuple and clippy complained it was too complicated. The reason I didn't make the fields public is to make the fact that these are different types clear (i.e., you do different things with each).

But whatever is sensible to you is fine with me. The nature of the type already makes it impossible to access the fields after you call run. I'll just make the fields public.